printstraat/backend/plugin/src/main/kotlin/com/ritense/valtimo/backend/plugin/PrintstraatPlugin.kt

```diff
@@ -17,14 +17,92 @@
 package com.ritense.valtimo.backend.plugin
 
 import com.ritense.plugin.annotation.Plugin
+import com.ritense.plugin.annotation.PluginAction
+import com.ritense.plugin.annotation.PluginActionProperty
 import com.ritense.plugin.annotation.PluginProperty
+import com.ritense.plugin.domain.ActivityType
+import com.ritense.resource.service.TemporaryResourceStorageService
+import com.ritense.valtimo.implementation.commutr.printstraat.dto.PrintstraatBodyDto
+import org.camunda.bpm.engine.delegate.DelegateExecution
+import org.springframework.http.HttpStatus
+import org.springframework.web.client.HttpServerErrorException
+import org.springframework.web.reactive.function.client.WebClient
+import java.net.URI
+import java.util.Base64
 
 @Plugin(
     key = "printstraat",
     title = "Printstraat Plugin",
     description = "Send a file to the printstraat in Utrecht with the Printstraat plugin"
 )
-class PrintstraatPlugin{
-    @PluginProperty(key = "zaaknummer", secret = false)
-    lateinit var zaaknummer: String
+class PrintstraatPlugin(
+    val temporaryResourceStorageService: TemporaryResourceStorageService
+) {
+    @PluginProperty(key = "url", secret = false)
+    lateinit var url: URI
+
+    @PluginProperty(key = "token", secret = true)
+    lateinit var token: String
+
+
+    @PluginAction(
+        key = "send-document-to-printstraat",
+        title = "Send document to Printstraat",
+        description = "Sends an document to the printstraat",
+        activityTypes = [ActivityType.SERVICE_TASK_START]
+    )
+    fun sendDocumentToPrintstraat(
+        execution: DelegateExecution,
+        @PluginActionProperty zaaknummerVariable: String?,
+        @PluginActionProperty tempFileIdVariable: String?
+    ) {
+        val tempFileId = execution.getVariable(tempFileIdVariable) as String? ?: execution.getVariable(
```

Comment:
Even voor mijn begrip. Of iemand zet deze variabel expliciet of iemand gebruikt een tool/plugin waarin de variabel automatisch wordt gezet? Als geen van beide waar is, is het dan handig om een exception te throwen? of is de standaard exceptie al duidelijk genoeg?